tests/MiddlewareTest.php

```diff
@@ -73,6 +73,22 @@ public function testAddsRetrySubscriber()
         $this->assertCount(0, $mock);
     }
 
+    public function testAddInvocationId()
+    {
+        $list = new HandlerList();
+        $mock = function ($command, $request) use (&$req) {
+            $req = $request;
+            return \GuzzleHttp\Promise\promise_for(
+                new Result(['@metadata' => ['statusCode' => 200]])
+            );
+        };
+        $list->setHandler($mock);
+        $list->prependSign(Middleware::invocationId());
+        $handler = $list->resolve();
+        $handler(new Command('foo'), new Request('GET', 'http://exmaple.com'));
+        $this->assertTrue($req->hasHeader('aws-sdk-invocation-id'));
```

Comment:
I think this assertion should be in the `$mock` function rather than inspecting a symbol that is updated by reference.